### PR TITLE
feat(iast): keep span if vulnerability is reported

### PR DESCRIPTION
**Pull request by avara1986**
## Description
Keep span if vulnerability is reported

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

File: ddtrace/appsec/iast/processor.py [hunk 1]
```diff
@@ -46,4 +48,8 @@ def on_span_finish(self, span):
         if data:
             span.set_tag_str(IAST_JSON, json.dumps(attr.asdict(data)))
 
+            span.set_tag(MANUAL_KEEP_KEY)
+            if span.get_tag(ORIGIN_KEY) is None:
+                span.set_tag_str(ORIGIN_KEY, "appsec")
```
Comment (gnufede) [inline ddtrace/appsec/iast/processor.py:53]: nit: maybe `"appsec"` should be a constant to avoid spelling mistakes

**Review by gnufede — DISMISSED** [review]

**Review by juanjux — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

File: ddtrace/appsec/iast/processor.py [hunk 2]
```diff
@@ -46,4 +49,8 @@ def on_span_finish(self, span):
         if data:
             span.set_tag_str(IAST_JSON, json.dumps(attr.asdict(data)))
 
+            span.set_tag(MANUAL_KEEP_KEY)
```
Comment (brettlangdon) [inline ddtrace/appsec/iast/processor.py:52]: this will keep the entire trace, is that expected? or do you just want to keep the individual span?

Comment (avara1986) [reply ddtrace/appsec/iast/processor.py:52]: Yes it is, we wanna keep the entire trace :+1:

Comment (brettlangdon) [reply ddtrace/appsec/iast/processor.py:52]: Great, thanks for confirming, I just wanted to be sure since it was listed as "keep span if..."

**Pull request merged**